File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const count = new Map();
        const freq = Array.from({length: nums.length + 1}, () => []);

        for (const num of nums) {
            if (!count[num]) {
                count[num] = 1;
            } else {
                count[num]++;
            }
        }

        for (const num in count) {
            freq[count[num]].push(num);
        }

        const output = [];

        for (let i = freq.length-1; i > 0; i--) {
            for (const num of freq[i]) {
                output.push(num);

                if (output.length === k) {
                    return output;
                }
            }
        }

        return output;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        res = nums[0]

        l, r = 0, len(nums)-1

        while l <= r:
            m = l + (r-l) // 2

            if nums[m] > nums[r]:
                l = m + 1
            else:
                r = m - 1

            res = min(res, nums[m])

        return res